### PR TITLE
Remove pendingChildren

### PR DESCRIPTION
**Pull request by peppy**
Seems to work as expected!

### CODE REVIEW

File: osu.Framework/Graphics/Containers/Container.cs [hunk 1]
```diff
@@ -56,22 +57,31 @@ public Container(LifetimeList<T> lifetimeList = null)
             };
         }
 
+        private Game game;
+
+        protected Task LoadComponentAsync(Drawable component, Action<Drawable> onLoaded = null) => component.LoadAsync(game, Clock, onLoaded);
```
Comment (Tom94) [inline osu.Framework/Graphics/Containers/Container.cs]: How about `LoadChildAsync`? Also, should throw exception if `component` is not a child. `Contains` should be pretty efficient.

Comment (peppy) [inline osu.Framework/Graphics/Containers/Container.cs]: Wasn't sure if we should use Child when the target is not yet added as a child. I'm fine with it if you are.

Comment (peppy) [inline osu.Framework/Graphics/Containers/Container.cs]: Keep in mind that this doesn't force the adding as a child part; that's up to the callback.

Comment (Tom94) [inline osu.Framework/Graphics/Containers/Container.cs]: In that case, shouldn't the `Clock` also be part of the callback?

I propose to always add as child inside `LoadChildAsync`. If someone wants to _not_ add as child, they should call `LoadAsync` (can no longer be internal) on the `Game` and provide their own `Clock`.

File: osu.Framework/Graphics/Containers/Container.cs [hunk 2]
```diff
@@ -56,22 +57,31 @@ public Container(LifetimeList<T> lifetimeList = null)
...
             };
 
-            if (pendingChildren != null)
+            try
             {
-                AddInternal(pendingChildren);
-                pendingChildren = null;
+                internalChildren.Update(Clock.TimeInfo);
+            }
+            catch (ObjectDisposedException)
```
Comment (Tom94) [inline osu.Framework/Graphics/Containers/Container.cs]: I really don't like this. Can we go over this in more detail on saturday when my deadline is over?

**Pull request merged**